src/main/java/apoc/util/Util.java

```diff
@@ -525,4 +531,20 @@ public static boolean transactionIsTerminated(GraphDatabaseAPI db) {
 
         return false;
     }
+
+    public static List<WeightedNodeResult> toWeightedNodeResultFromLegacyIndex(LegacyIndexHits hits, GraphDatabaseService db){
```

Comment:
I don't think these should go into util but stay in the index package, perhaps in an IndexUtils there.
Or even having KernelAPI return a stream of `WeightedNodeResult` or `WeightedRelationshipResult`